### ISSUE TITLE
Create Java API for Lumo Utility Classes

### ISSUE THREAD

**Issue by tarekoraby** [vaadin/flow-components#2410]
### Describe your motivation

Consider creating a Java API for Lumo Utility Classes. Currently, if one wants to make use of those classes in a Flow app, one would have to do something like
```
addClassNames("flex", "flex-col", "gap-l", "mb-s", "mt-0", "text-s");
```
But this is suboptimal because (1) users could make typos in the String literals, (2) users don't necessarily know all the available class names, and (3) users can unknowingly add classess that cancel other classes (like adding `border-0` followed by `border`). 

### Describe the solution you'd like

I suggest adding `addUtilityClasses(UtilityClass)` and `removeUtilityClasses(UtilityClass)` methods to `com.vaadin.flow.component.HasStyle` interface (where `UtilityClass` is an Enum).

**Comment by tarekoraby** [comment]
If there’s a green light on this, I’d be happy to contribute a PR.

**Comment by web-padawan** [comment]
Please note, the utility classes are not meant to be used by Vaadin components, but rather by application views. Therefore, I would object adding this API to `HasStyle`. IMO this logic should be placed in a completely separate interface.

**Comment by tarekoraby** [comment]
Why not use them with components? What is the downside?

**Comment by rolfsmeds** [comment]
They don't really _work_ with components systematically, because shadow dom, etc etc. Also take TextField as an example. What would applying a text-color utility class to it even mean? TextFields have four different text elements: label, value, error message and helper text. Which one(s) should the class apply to?

So while _some_ utility classes _happen_ to apply to certain components in the way you'd expect, that's a coincidence rather than by-design.

**Comment by tarekoraby** [comment]
Ok, I see. In that case, could we just add a `UtilityClass` enum somewhere (apart from `HasStyle`), and thereby enable devs to use that enum with the existing `HasStyle.addClassNames()`?

**Comment by sissbruecker** [comment]
Adding constants for the class names and then using the existing `HasStyle` methods would be good enough IMO.

This could also be done in a separate library for now to gauge interest and verify the benefits.

**Comment by tarekoraby** [comment]
> Adding constants for the class names and then using the existing HasStyle methods would be good enough IMO.

Agreed. So could I get a green light to add those constants under a new class? I suggest creating a new `LumoClasses` class under `com.vaadin.flow.component`. Is that ok?

That would considerably simplify styling for Flow users, I believe.
